lib/install.js

```diff
@@ -208,7 +208,7 @@ function Installer (where, dryrun, args) {
   this.todo = []
   this.progress = {}
   this.noPackageJsonOk = !!args.length
-  this.topLevelLifecycles = !args.length
+  this.topLevelLifecycles = !!args.length
```

Comment:
Your `preinstall` top level lifecycle shouldn't run if you run `npm install foobar`. It should only run if you're installing the top level package, eg `npm install`.